lonboard/experimental/traits.py

```diff
@@ -101,3 +102,71 @@ def validate(
 
         self.error(obj, value)
         assert False
+
+
+class NormalAccessor(Union[TraitType, FixedErrorTraitType]):
+    """
+        A representation of a deck.gl normal accessor
+        
+        Acceptable inputs:
+        - A numpy ndarray with two dimensions: The size of the second dimension must be 3 i.e. `(N,3)`
+        - a pyarrow `FixedSizeListArray` or `ChunkedArray` containing `FixedSizeListArray`s 
+        where the size of the inner fixed size list 3.
+    """
+    default_value = [0,0,1]
+    info_text = (
+        "List representing normal of each object, in [nx, ny, nz]. or numpy ndarray or "
+        "pyarrow FixedSizeList representing the normal of each object, in [nx, ny, nz]"
+    )
+
+    def __init__(
+        self: TraitType,
+        *args,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.tag(sync=True, **NORMAL_SERIALIZATION)
+        
+    def validate(
```

Comment:
I'd recommend writing some tests first, which will make it easier to sort out the logic in this function. You can find some related tests [here](https://github.com/developmentseed/lonboard/blob/16d4c3ce41fbd540585b4ef3e5395cee4038e8ee/tests/test_traits.py).

The purpose of this validation function is to take whatever user input and both validate it _and_ conform to a standard expectation of how we want to store it internally.

To align with deck.gl and other traits, we should allow a list with three scalar values, or a numpy array, or a pyarrow FixedSizeListArray. 

But here the first thing you check is `value.shape`. Which won't exist for a list type. So it'll crash

```
In [1]: l = [1, 23, 50]

In [2]: l.shape
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[2], line 1
----> 1 l.shape

AttributeError: 'list' object has no attribute 'shape'
```

Hence why adding tests first might help